src/components/Card.js

```diff
@@ -1,37 +1,47 @@
 export default class Card {
-  constructor({ title, url }, cardSelector, handleImageClick) {
-    this._data = { title, url };
-    this._name = title;
-    this._link = url;
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClicks
+  ) {
+    this._name = name;
+    this._link = link;
+    this._id = _id;
+    this.isLiked = isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClicks = handleLikeClicks;
   }
 
   _setEventListeners() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeIcon(this);
-      });
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeClicks(this);
+    });
 
-    this._cardElement
-      .querySelector(".card__delete-button")
-      .addEventListener("click", () => {
-        this._handleDeleteCard();
-      });
+    this._deleteButton.addEventListener("click", () => {
+      this._handleDeleteClick(this);
+    });
 
     this._cardImage.addEventListener("click", () => {
-      this._handleImageClick(this._data);
+      this._handleImageClick({
+        name: this._name,
+        link: this._link,
+      });
     });
   }
 
-  _handleLikeIcon() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
```

Comment:
Needs correction: renderLikes method should be private and should be used only inside the class. For updating the likes state you should create a setter which will set the state and render the button according to it:

```js
// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 
```